### PR TITLE
Employee details screen refactoring

### PR DESCRIPTION
**Pull request by antonm76**
Employee details screen is refactored to have a full screen scroll and a refresh control

### CODE REVIEW

File: mobile/src/employee-details/employee-details-screen.tsx [hunk 1]
```diff
@@ -2,42 +2,124 @@ import React, { Component } from 'react';
...
+
+    //----------------------------------------------------------------------------
+    public shouldComponentUpdate(nextProps: EmployeeDetailsProps & EmployeeDetailsDispatchProps & NavigationScreenProps, nextState: EmployeeDetailsState) {
+
+        if (this.state.width !== nextState.width || this.state.height !== nextState.height) {
+            return true;
+        }
+
+        const departments = this.props.departments;
+        const nextDepartments = nextProps.departments;
```
Comment (santee) [inline mobile/src/employee-details/employee-details-screen.tsx]: probably not relevant but, if nextDepartments becomes null or undefined suddenly, the component will not rerender as it returns true.

File: mobile/src/employee-details/employee-details.tsx [hunk 2]
```diff
@@ -192,15 +192,15 @@ export class EmployeeDetailsImpl extends Component<EmployeeDetailsProps & Employ
...
+                <View style={this.props.layoutStylesChevronPlaceholder}/>
                 <View>
                     <Chevron/>
                     <View style={layoutStyles.avatarContainer}>
                         <Avatar photoUrl={employee.photoUrl} imageStyle={{ borderWidth: 0 }}
                                 style={{ borderWidth: 3 }}/>
                     </View>
                 </View>
-                <ScrollView style={layoutStyles.scrollView} alwaysBounceVertical={false}>
+                <View style={layoutStyles.scrollView}>
```
Comment (santee) [inline mobile/src/employee-details/employee-details.tsx]: weird naming issues? :)

File: mobile/src/employee-details/employee-details-screen.tsx (hunk 1 above)
Comment (antonm76) [reply mobile/src/employee-details/employee-details-screen.tsx]: a86721f8d043aea5fdf571a64c75b7d7cd304bc7

File: mobile/src/employee-details/employee-details.tsx (hunk 2 above)
Comment (antonm76) [reply mobile/src/employee-details/employee-details.tsx]: 64f3d730ad90760184b750d1afa164f782cd1651

**Pull request merged**